### PR TITLE
Expire miscellaneous deprecations from 3.5

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

These are not handled from the list in https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.5.0.html#miscellaneous-deprecations

```
- ``Collection.__init__`` previously ignored *transOffset* without *offsets* also
  being specified. In the future, *transOffset* will begin having an effect
  regardless of *offsets*. In the meantime, if you wish to set *transOffset*,
  call `.Collection.set_offset_transform` explicitly.
```
I do not follow.

```
- ``dviread.PsfontsMap`` now raises LookupError instead of KeyError for missing
  fonts.
```
Should not have been in deprecation?

```
- ``axisartist.clip_path`` is deprecated with no replacement.
```
~Cannot find it.~
## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -5399,12 +5391,11 @@ def fill_betweenx(self, y, x1, x2=0, where=None,
     # Once this deprecation elapses, also move vmin, vmax right after norm, to
     # match the signature of other methods returning ScalarMappables and keep
     # the documentation for *norm*, *vmax* and *vmin* together.
-    @_api.make_keyword_only("3.5", "aspect")
     @_preprocess_data()
     @_docstring.interpd
-    def imshow(self, X, cmap=None, norm=None, aspect=None,
+    def imshow(self, X, cmap=None, norm=None, *, aspect=None,
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:5396]: Need to apply the change from the above comment.

**Comment by oscargus** [comment]
Thanks! Sorted!

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -2881,10 +2881,9 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
 
           stem([locs,] heads, linefmt=None, markerfmt=None, basefmt=None)
 
-        The *locs*-positions are optional. The formats may be provided either
-        as positional or as keyword-arguments.
-        Passing *markerfmt* and *basefmt* positionally is deprecated since
-        Matplotlib 3.5.
+        The *locs*-positions are optional. *linefmt* can be provided as
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:2884]:
```suggestion
        The *locs*-positions are optional. *linefmt* may be provided as
```

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -5407,14 +5399,12 @@ def fill_betweenx(self, y, x1, x2=0, where=None,
     # Once this deprecation elapses, also move vmin, vmax right after norm, to
     # match the signature of other methods returning ScalarMappables and keep
     # the documentation for *norm*, *vmax* and *vmin* together.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:5401]: And delete comment?

**Review by anntzer — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

**Pull request merged**